### ISSUE TITLE
Is this script abandoned?

### ISSUE THREAD

**Issue by drivebass** [wojodesign/simplecart-js#420]
It's been quite a while since February, the month the developers had announced on their twitter account that they will be adding the mail check function etc. Just wondering if this project has been forgotten in time.

<blockquote class="twitter-tweet"><p><a href="https://twitter.com/EdsonHong">@EdsonHong</a> Thanks Edson! Great idea, we are working on this and many other things, real booked though. Won&#39;t be done till February at least.</p>&mdash; simpleCart(js) (@simplecartjs) <a href="https://twitter.com/simplecartjs/statuses/273593437222232064">November 28, 2012</a></blockquote>

<script async src="//platform.twitter.com/widgets.js" charset="utf-8"></script>

**Comment by Malstrom** [comment]
you're right, it's a shame too because I find it is the best free shopping cart available on the internet

**Comment by Malstrom** [comment]
You use that?

**Comment by drivebass** [comment]
@Malstrom Yes, it is absolutely the most straight forward shopping cart around the web. I was pretty surprised to find out it was free when i discovered it. I guess i would pay for a complete final version. Probably the developers had projects to finish and no time for this. I've been using it for quite a while now, solved some problems with the mail check out with php and tweaked the code a lot to the needs of a project. Haven't test live yet though.

**Comment by Malstrom** [comment]
@drivebass 
thanks!!!
i follow your suggestion and add this method on checkout of simplecart for my app :)

**Comment by codeofsumit** [comment]
was just trying to build a shop when i saw like 3 bugs in the first hour and somehow no activity from the devs. Very sad. Gonna use something else or (worst case) build my own

**Comment by drivebass** [comment]
@Sumit8 Have you checked bug issues here? Because most of the bugs regarding the basic functions are pretty much solved.

**Comment by codeofsumit** [comment]
@drivebass Well, insert the very basic cart view and check your favourite dev-tool. SimpleCart creates the "headerRow"-DIV and inserts elements like the "remove"-DIV into it (which is a mistake). Custom views' HTML get inserted as a string into the elements attribute (flaw # 2). That was in the latest built (downloaded today).
Do you guys know any simple and very flexible and customizable cart-system out there that has an active development base? PHP would be ok as well i guess.

**Comment by drivebass** [comment]
@Sumit8 I don't quite get what your issue is, maybe some code would help. Although this script was supposed to keep things basic and simple i'm sure you can extend it to your needs. I'm not aware of any SIMPLE cart solution, i think with simplecart it's the simplest it gets, aside the absence of support from the developers, but after all it's free and we shouldn't complain:) I would really buy this if it was a complete script with full support.

**Comment by codeofsumit** [comment]
@drivebass it's a total different story if it'd had support. I'm willing to buy a cart script as well - even expensive ones. But they need to be very flexible - and most of em arent (even if they claim to be).
Simple Cart is flexible and yes, i could fix the issues myself but knowing there is no support is a no brainer to not use it for a new shop.

**Comment by drivebass** [comment]
@Sumit8 I guess our best bet is to try and improve this script ourselves instead of looking for another solution or even worst to try to create our own. This script is really too clever and simple to ignore it. I really wonder why the developers have left it idle. Anyway. I reviewed your issue again and i figured out what the problem is:

When the cart Style is set to "table" the script must create the individual cells for each column even when the label is set to false so it will output an empty cell in the header otherwise the table layout would break. But when the cartStyle is set to "div" there is no conditional logic in the script to prevent it from creating the extra div's which in this case are not essential for layout. The code that handles this is on line 648. I added an if statement that if the label is set to false no extra divs are appended in the headerRow. Tested and worked fine. I guess it needs some extra condition in the case the style is to table, because with this it will break the table layout.

```
// append the header cell
if (column.label != false) {                    
   header_container.append(
       simpleCart.$create(TH).addClass(klass).html(label)
   );
 }   
```
